crates/driver/src/domain/competition/auction.rs

```diff
@@ -142,66 +143,56 @@ impl Auction {
+                    order::Partial::Yes { available } => available.0,
+                    order::Partial::No => target,
                 };
 
-                let remaining = Remaining::from_order(&RemainingOrder {
-                    kind: match order.side {
-                        order::Side::Buy => model::order::OrderKind::Buy,
-                        order::Side::Sell => model::order::OrderKind::Sell,
-                    },
-                    buy_amount: order.buy.amount.0,
-                    sell_amount: order.sell.amount.0,
-                    fee_amount: order.fee.user.0,
-                    executed_amount: match order.partial {
-                        order::Partial::Yes { executed } => executed.0,
-                        order::Partial::No => 0.into(),
-                    },
-                    partially_fillable: match order.partial {
-                        order::Partial::Yes { .. } => true,
-                        order::Partial::No => false,
-                    },
-                })
-                .context("Remaining::from_order")?;
-                let sell = remaining
-                    .remaining(order.sell.amount.0)
-                    .context("remaining_sell")?;
-                let fee = remaining
-                    .remaining(order.fee.user.0)
-                    .context("remaining_fee")?;
-                sell.checked_add(fee).context("add sell and fee")
-            }
+                let sell = util::math::mul_ratio(order.sell.amount.0, available, target)?;
+                let fee = util::math::mul_ratio(order.fee.user.0, available, target)?;
 
-            let max_fill = match max_fill(order) {
-                Ok(balance) => balance,
-                Err(err) => {
-                    let reason =
-                        observe::OrderExcludedFromAuctionReason::CouldNotCalculateRemainingAmount(
-                            &err,
-                        );
-                    observe::order_excluded_from_auction(order, reason);
+                sell.checked_add(fee)
+            }
+            let max_sell = match max_sell(order) {
+                Some(amount) => amount,
+                None => {
+                    observe::order_excluded_from_auction(
+                        order,
+                        observe::OrderExcludedFromAuctionReason::CouldNotCalculateMaxSell,
+                    );
                     return false;
                 }
             };
 
-            let used_balance = match order.is_partial() {
-                true => {
-                    if *remaining_balance == 0.into() {
-                        return false;
-                    }
-                    max_fill.min(*remaining_balance)
-                }
-                false => {
-                    if *remaining_balance < max_fill {
-                        return false;
-                    }
-                    max_fill
-                }
+            let used_sell = match order.partial {
```

Comment:
Is this really used_sell? Feels more like `remaining_sell` since it's taking `min(remaining_balance)`

Reply:
In this context, I meant it as the “used sell balance”

Reply:
But how is this the "used sell balance"? If I understand the code correctly it is either
1. The min of remaining balance and remaining sell amount
2. Zero if fok and remaining balance is not enough

So this is the "allocated sell balance" of this order for the upcoming auction?

Maybe some comments in the code would help clarify the semantics given we opt for short variable names.

Reply:
> So this is the "allocated sell balance" of this order for the upcoming auction?

Yes

Reply:
> Maybe some comments in the code would help clarify the semantics given we opt for short variable names.

I would rather change the code so it is clearer what is going on.

Reply:
Rewrote it a bit in https://github.com/cowprotocol/services/commit/fd15cb145ee8f0eb628492d86703f2baa5013290, let me know if things are clearer or not, and if I should still add comments explaining anything.